lib/clean_architecture/interfaces/success_payload.rb

```diff
@@ -7,7 +7,7 @@ module Interfaces
     module SuccessPayload
       extend Duckface::ActsAsInterface
 
-      def data_hash
+      def data
```

Comment:
That's true. I have a couple of thoughts about this.

1. When releasing a gem again in future, we should look at keeping the project at pre 1.0.0 for a longer period of time until the API becomes stable. https://semver.org/#spec-item-4  This will then make it easier to use semantic versioning to add new features and deprecate old ones. It is then much easier for the consuming applications to take up the new functionality whenever required.

2. Like with any library, if it is wide use, then there is no avoiding the nature of updating it. If we notice that it is such a critical part of our toolbox, we should proactively spend more time making sure that the API is suitable that it doesn't require constant change.

I don't feel that strongly about enforcing a strict policy for either of these points, I just think that it's something that we should push for with each commit to make the API something we agree on, consistent and robust for its intended purpose.

In relation to this change in particular, the method can now return either a `Hash` or an `Array` so I wanted the name to not lie about the return type.

Reply:
@lgsanjos I'm merging this now, but let's continue this on this point - I'd like to resolve it.